### ISSUE TITLE
Username part of membership

### ISSUE THREAD

**Issue by eeemarv** [dimitrydhondt/inner-gerbil#26]
In parties:  
```javascript
    alias: $s.string('Handle the party wants to be known by.'),
```

So this is actually a username. But it's unique system-wide (or should be). 

I think the username is the responsability of the group. In other words, the username is a property of partyrelations. (the membership)

Then,
* Usernames don't have to be unique system-wide. It doesn't make sense a certain username gets blocked because of a user with the same name which is not part of your group.
* Groups can decide themselves the style of the username, how much they reveal of the identity of the person and if users can set their usernames themselves.

**Comment by dimitrydhondt** [comment]
In my world it was not unique. I'm not sure if it makes sense to have a different alias in a different group.
What do you think ?

Verzonden via Yahoo Mail op Android 
 
  Op za, nov. 21, 2015 om 12:45 schreef martti<notifications@github.com>:   
In parties:

 alias: $s.string('Handle the party wants to be known by.'),

So this is actually a username. But it's unique system-wide (or should be). 

I think the username is the responsability of the group. In other words, the username is a property of partyrelations. (the membership)

Then,
   
   - Usernames don't have to be unique system-wide. It doesn't make sense a certain username gets blocked because of a user with the same name which is not part of your group.
   - Groups can decide themselves the style of the username, how much they reveal of the identity of the person and if users can set their usernames themselves.

**Comment by eeemarv** [comment]
It can be useful, 'handle the party wants to be known by' expresses a wish, it is not identification yet. It can be a hint to the admins of a group. Then it's up to the admins of the group if they accept the wish or that they give the person themselves a username or alias in accordance to the culture of the group. The point is, finally the group has to be in control.

**Comment by renardeau** [comment]
Both point of views make sense to me, but aren't we mixing stuff?

@eeemarv 
That unique username or alias a group is giving to its members, isn't that the letscode you are referring to in issue #24 ?

Is the alias in parties to be understood as a 'nickname'?

My understanding:
* leave the alias in parties
* add a letscode/alias/... as unique membership id within a group to partyrelations.

**Comment by eeemarv** [comment]
Some groups might prefer to be in control of both the nickname and the letscode. This is the case in letsgroup where I'm member off. 

In our group:
* The letscode is the primal reference. It won't change during the lifetime of the account. Except in the case our group will know exceptional growth, then we will prefix all existing accounts with a zero.
* The username has a specific format to the culture of our group. We don't want users to chose whatever fantasy name (like i.e _eeemarv_). And we want to protect the privacy of our users so we don't use the full name, but we want them to have a link to the real life, so we use the first name here. Maybe with some initials of the last name to ensure uniqueness. The username may change during the lifetime of the account but not too often in order to avoid confusion. Users can not change their username themselves, they always have to ask the admins. i.e. last week someone asked me to change her name from Ann to Anneke, because she was more known as Anneke.

In short in our group the administration wants to be in control of both letscodes and usernames. 

Other groups might have other an other opinion. I.e. our neighbouring group uses the fullname as username. Everyone has to show their full name in this group. In an other group I saw fantasy names being used. Some groups that are not using eLAS don't use letscodes at all. The names and (lets)codes are part of the culture of the group.

What is more important? The culture of the group or that someone is identified everywhere with the same username? Definitely the culture of the group imo. It might in practice also be just a few people who have multiple memberships.

**Comment by dimitrydhondt** [comment]
Move this discussion to a functional PM tool ? https://basecamp.com/ ?

**Comment by dimitrydhondt** [comment]
I've added 'code' to partyrelations and documented that it is up to the group to decide if they use full name always, allow aliases (or not), or use group-bound code to identify users in the UI.
Internally we obviously stick to permalinks.

**Comment by eeemarv** [comment]
And the functional discussion? In general I would like at least to keep the functionality we have today. Like, control of the group over the username.
